### PR TITLE
Feature/answer controllers

### PR DESCRIPTION
**Pull request by AdrieneTZ**
**This branch has four features:**
- Get a specific answer with its id
- Create a new answer to the specific question
- Edit the specific answer, and the current user can only edit the answer he posted before
- Delete the specific answer, and the current user can only delete the answer he posted before

Fix an issue while testing `POST /api/v1/questions/:id/answers`. 
In `prisma/schema.prisma` `question` and `answer` content field, we set this attribute as `String`. This `String` is mapping to `MySQL varchar(191)`. As the content length is over 191 characters, database server would refuse to accept the content we sent.
To fix this issue, I add `@db.TEXT` after `String`. Please run `npx prisma migrate dev` after you merge the remote branch to your local branch.

Thank you for reviewing this PR. After merging this branch, we can start adding tests to our project. Yeah!

In commit `742c3d4`, I deleted user's password and returned all user data being changed approvalStatus.

### CODE REVIEW

File: controllers/answer-controller.js [hunk 1]
```diff
@@ -0,0 +1,203 @@
...
+
+      if (typeof content !== 'string') {
+        return res.status(400).json({
+          status: '400F',
+          message: 'Field: datatype of the content must be string.',
+        })
+      }
+
+      content = content.trim()
+      if (content.length > 500) {
```
Comment (seanlin1125) [inline controllers/answer-controller.js:57]: could refactor as `if (content?.trim().length > 500) ...`

File: controllers/answer-controller.js [hunk 2]
```diff
@@ -0,0 +1,203 @@
...
+        return res.status(406).json({
+          status: 'error',
+          message: 'Content length must be less than 500 characters.',
```
Comment (seanlin1125) [inline controllers/answer-controller.js:60]: I've checked about the usage of statusCode 406, it seems that it's more about the header format error
say if it should be application/json but client sent text/plain then we'll response 406
In this case the error is more like "400F" as we've discussed before?

File: controllers/answer-controller.js [hunk 3]
```diff
@@ -0,0 +1,203 @@
...
+
+      if (typeof content !== 'string') {
+        return res.status(400).json({
+          status: '400F',
+          message: 'Field: datatype of the content must be string.',
+        })
+      }
+
+      content = content.trim()
+      if (content.length > 500) {
```
Comment (seanlin1125) [inline controllers/answer-controller.js:128]: same as line 56-57, could consider combining together

File: controllers/answer-controller.js [hunk 4]
```diff
@@ -0,0 +1,203 @@
...
+        return res.status(406).json({
```
Comment (seanlin1125) [inline controllers/answer-controller.js:129]: same as line 58, could discuss whether 406 is appropriate or not

**Review by seanlin1125 — COMMENTED** [review]
Great Job!
Could totally see your mindfulness when coding
check about the status 406 and we're good to go
would merge to the main once we're on the same page of 406
let me know when you're available to discuss

File: controllers/answer-controller.js (hunk 2 above)
Comment (AdrieneTZ) [reply controllers/answer-controller.js:60]: Sure, we have to discuss about using `406` or `400F`. As for choosing to use `406` or `400F`, I was thinking, is there too much 400F in our controllers, which would also make our frontend partners hard to deal with error issues? So I choose `406` in this case.

**Comment by AdrieneTZ** [comment]
Refactor all `400` error statuses for our frontend partners to debug more easily.

**Review by seanlin1125 — APPROVED** [review]
LGTM~thx a lot!

**Pull request merged**